src/peft/tuners/waveft/waverec2.py

```diff
@@ -0,0 +1,286 @@
+import functools
+import typing
+import warnings
+from collections.abc import Callable, Sequence
+from functools import partial
+from typing import Any, List, Literal, NamedTuple, Optional, Protocol, Union, cast, overload
+
+import numpy as np
+import pywt
+import torch
+from typing_extensions import ParamSpec, TypeVar, TypeAlias, Unpack
+
+# --- From constants.py ---
```

Comment:
If `constants.py` is referring to a file in another repo, please add a permalink to that file instead, otherwise it's hard to know what you're referring to.